src/core/lib/gpr/tls.h

```diff
@@ -59,7 +59,7 @@ class TlsTypeConstrainer {
 
 #ifdef GPR_STDCPP_TLS
 #define GPR_THREAD_LOCAL(type) \
-  thread_local grpc_core::TlsTypeConstrainer<type>::Type
+  thread_local typename grpc_core::TlsTypeConstrainer<type>::Type
```

Comment:
how come only this one needed "typename"?

Reply:
I likely need to propagate it... late night hacking to get even the basics compiling again.

Reply:
done

Reply:
cool, but just for my edification: why was that change needed at all?

Reply:
I'm using this type in a template, and C++ says given Foo<T>::Bar I can't know if Bar is a value or a type for all T, but I need to at that point in the code so I need to say it.

Reply:
Thanks! Good to know.